handler/authentication.go

```diff
@@ -0,0 +1,224 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/labstack/echo"
+	"github.com/slack-go/slack"
+	"github.com/webuild-community/core/model"
+	"go.uber.org/zap"
+	"gorm.io/gorm"
+)
+
+type AuthorizeHandler struct {
+	clientID     string
+	clientSecret string
+	logger       *zap.Logger
+	db           *gorm.DB
+	slack        *slack.Client
+}
+
+func NewAuthorizeHandler(
+	e *echo.Echo,
+	logger *zap.Logger,
+	db *gorm.DB,
+	slack *slack.Client,
+) *AuthorizeHandler {
+	clientID := os.Getenv("GITHUB_CLIENT_ID")
+	if len(clientID) == 0 {
+		logger.Fatal("GITHUB_CLIENT_ID is not set")
+	}
+	clientSecret := os.Getenv("GITHUB_CLIENT_SECRET")
+	if len(clientSecret) == 0 {
+		logger.Fatal("GITHUB_CLIENT_SECRET is not set")
+	}
+
+	handler := &AuthorizeHandler{
+		clientID:     os.Getenv("GITHUB_CLIENT_ID"),
+		clientSecret: os.Getenv("GITHUB_CLIENT_SECRET"),
+		logger:       logger,
+		db:           db,
+		slack:        slack,
+	}
+
+	e.GET("/callback/github/auth", handler.handleGithubCallback)
+
+	return handler
```

Comment:
k cần return handler nha e, e New giống bên này là đc https://github.com/webuild-community/core/blob/develop/handler/event.go

Reply:
Roger that, new version is updated!